File: jphp-javafx-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXWebEngine.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.scene.web.WebEngine;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import org.w3c.dom.Document;
import php.runtime.Memory;
import php.runtime.annotation.Reflection.Abstract;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

@Abstract
@Name(JavaFXExtension.NAMESPACE + "UXWebEngine")
public class UXWebEngine extends BaseWrapper<WebEngine> {
    interface WrappedInterface {
        @Property Document document();
        @Property boolean javaScriptEnabled();
        @Property String location();
        @Property String title();

        @Property String userStyleSheetLocation();

        void load(String url);
        void loadContent(String content);
        void loadContent(String content, String contentType);

        void reload();
    }

    public UXWebEngine(Environment env, WebEngine wrappedObject) {
        super(env, wrappedObject);
    }

    public UXWebEngine(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public Memory executeScript(Environment env, String script) {
        return Memory.wrap(env, getWrappedObject().executeScript(script));
    }
}

File: jphp-javafx-ext/src/main/resources/JPHP-INF/sdk/ext/javafx/UXWebEngine.php
<?php
namespace ext\javafx;

use ext\xml\DomDocument;

/**
 * Class UXWebEngine
 * @package ext\javafx
 */
abstract class UXWebEngine
{
    /**
     * @readonly
     * @var DomDocument
     */
    public $document;

    /**
     * @var bool
     */
    public $javaScriptEnabled;

    /**
     * @var string
     */
    public $userStyleSheetLocation;

    /**
     * @readonly
     * @var string
     */
    public $location;

    /**
     * @readonly
     * @var string
     */
    public $title;

    /**
     * @param string $url
     */
    public function load($url) {}

    /**
     * @param string $content
     * @param string $contentType (optional)
     */
    public function loadContent($content, $contentType) {}

    /**
     * ...
     */
    public function reload() {}

    /**
     * @param string $script
     * @return mixed
     */
    public function executeScript($script) {}
}